data-loader/core/src/main/java/com/scalar/db/dataloader/core/dataexport/validation/ExportOptionsValidator.java

```diff
@@ -0,0 +1,111 @@
+package com.scalar.db.dataloader.core.dataexport.validation;
+
+import com.scalar.db.api.Scan;
+import com.scalar.db.api.TableMetadata;
+import com.scalar.db.common.error.CoreError;
+import com.scalar.db.dataloader.core.ScanRange;
+import com.scalar.db.dataloader.core.dataexport.ExportOptions;
+import com.scalar.db.io.Key;
+import java.util.LinkedHashSet;
+import java.util.List;
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+
+/**
+ * A validator for ensuring that export options are consistent with the ScalarDB table metadata and
+ * follow the defined constraints.
+ */
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public class ExportOptionsValidator {
+
+  /**
+   * Validates the export request.
+   *
+   * @param exportOptions The export options provided by the user.
+   * @param tableMetadata The metadata of the ScalarDB table to validate against.
+   * @throws ExportOptionsValidationException If the export options are invalid.
+   */
+  public static void validate(ExportOptions exportOptions, TableMetadata tableMetadata)
+      throws ExportOptionsValidationException {
+    LinkedHashSet<String> clusteringKeyNames = tableMetadata.getClusteringKeyNames();
+    ScanRange scanRange = exportOptions.getScanRange();
+
+    // Validate projection columns
+    validateProjectionColumns(tableMetadata.getColumnNames(), exportOptions.getProjectionColumns());
+
+    // Validate sort orders
+    if (!exportOptions.getSortOrders().isEmpty()) {
+      for (Scan.Ordering sort : exportOptions.getSortOrders()) {
+        validateClusteringKey(clusteringKeyNames, sort.getColumnName());
+      }
+    }
+
+    // Validate scan start key
+    if (scanRange.getScanStartKey() != null) {
+      validateClusteringKey(clusteringKeyNames, scanRange.getScanStartKey());
+    }
+
+    // Validate scan end key
+    if (scanRange.getScanEndKey() != null) {
+      validateClusteringKey(clusteringKeyNames, scanRange.getScanEndKey());
+    }
+  }
+
+  /**
+   * Checks if the provided clustering key is valid for the ScalarDB table.
+   *
+   * @param clusteringKeyNames The set of valid clustering key names for the table.
+   * @param key The ScalarDB key to validate.
+   * @throws ExportOptionsValidationException If the key is invalid or not a clustering key.
+   */
+  private static void validateClusteringKey(LinkedHashSet<String> clusteringKeyNames, Key key)
+      throws ExportOptionsValidationException {
+    if (clusteringKeyNames == null) {
+      return;
+    }
+    String columnName = key.getColumnName(0);
```

Comment:
Sorry, why are you only checking the first column?

Reply:
Jishnu is off for a bit so I took over this task and updated the code. I have fixed the validation and completely rewrote the unit tests to cover all scenarios.

- For the partition key(s) making sure all keys are provided and in order
- For clustering keys, make sure the order or partial order is correct. (full key A,B,C -->. so you can call with (A), (A,B), (A,B,C))

This fix/problem exposes a limitation in the current data loader where you can only call the data loader like this

`./data-loader export -pk <KEY=VALUE>`.   This will be fixed so you can do `<KEY=VALUE>,<KEY=VALUE>`. This code is coming with the next PR so it will already be fixed. But just FYI.

@inv-jishnu As mentioned, make sure to update the CLI arguments for export in the next PR to `protected List<ColumnKeyValue> partitionKeyValues;`